pkg/model/mask.go

```diff
@@ -115,6 +110,53 @@ func updateField(src, dst reflect.Value, fields []string) {
 	}
 }
 
+// updateSliceField this function updates a destination slice by either replacing a struct with a matching ID
+// or appending new values if no matching ID is found.
+func updateSliceField(srcField, dstField reflect.Value) {
+	appendOnly := func() bool {
+		el := srcField.Type().Elem()
+
+		if el.Kind() != reflect.Struct {
+			return true
+		}
+
+		_, ok := el.FieldByName("ID")
+
+		return !ok
+	}
+
+	if appendOnly() {
+		dstField.Set(reflect.AppendSlice(dstField, srcField))
+		return
+	}
+
+	// If found by id, replace the corresponding structure in the destination slice. Otherwise, append.
+	for i := 0; i < srcField.Len(); i++ {
+		srcID := srcField.Index(i).FieldByName("ID").Interface()
+		idx := findInDstField(srcID, dstField)
+
+		// Check if the "ID" field of the first element in the source slice
+		// matches the "ID" field of the current element in the destination slice.
+		if idx != -1 {
+			// If found, update the corresponding structure in the destination slice
+			dstField.Index(idx).Set(srcField.Index(i))
+			continue
+		}
+
+		dstField.Set(reflect.Append(dstField, srcField.Index(i)))
+	}
+}
+
+func findInDstField(srcID interface{}, dstField reflect.Value) int {
```

Comment:
`func updateSliceField` is relatively simple.

Can we add this the same way you did with `appendOnly` to avoid package level functions?

Simply `index := func(id any) int {}`?